compose/foundation/foundation/src/skikoMain/kotlin/androidx/compose/foundation/OnClick.skiko.kt

```diff
@@ -0,0 +1,444 @@
+ * within the component's bounds.
+ *
+ * It allows configuration based on a pointer type via [matcher].
+ * By default, matcher uses [PointerMatcher.Primary].
+ * [matcher] should declare supported pointer types (mouse, touch, stylus, eraser) by listing them and
+ * declaring required properties for them, such as: required button (primary, secondary, etc.).
+ *
+ * @param enabled Controls the enabled state. When `false`, [onClick], [onLongClick] or
+ * [onDoubleClick] won't be invoked
+ * @param matcher defines supported pointer types and required properties
+ * @param keyboardModifiers defines a condition that [PointerEvent.keyboardModifiers] has to match
+ * @param onLongClick will be called when user long presses on the element
+ * @param onDoubleClick will be called when user double clicks on the element
+ * @param onClick will be called when user clicks on the element
+ */
+@ExperimentalFoundationApi
+fun Modifier.onClick(
+    enabled: Boolean = true,
+    matcher: PointerMatcher = PointerMatcher.Primary,
+    keyboardModifiers: PointerKeyboardModifiers.() -> Boolean = { true },
+    onDoubleClick: (() -> Unit)? = null,
+    onLongClick: (() -> Unit)? = null,
+    onClick: () -> Unit
+) = composed {
+    Modifier.onClick(
+        enabled = enabled,
+        matcher = matcher,
+        keyboardModifiers = keyboardModifiers,
+        interactionSource = remember { MutableInteractionSource() },
+        onDoubleClick = onDoubleClick,
+        onLongClick = onLongClick,
+        onClick = onClick
+    )
+}
+
+/**
+ * Configure component to receive clicks, double clicks and long clicks via input only (no accessibility "click" event)
+ * within the component's bounds.
+ *
+ * It allows configuration based on a pointer type via [matcher].
+ * By default, matcher uses [PointerMatcher.Primary].
+ * [matcher] should declare supported pointer types (mouse, touch, stylus, eraser) by listing them and
+ * declaring required properties for them, such as: required button (primary, secondary, etc.).
+ *
+ * @param interactionSource [MutableInteractionSource] that will be used to emit
+ * [PressInteraction.Press] when this clickable is pressed. Only the initial (first) press will be
+ * recorded and emitted with [MutableInteractionSource].
+ * @param enabled Controls the enabled state. When `false`, [onClick], [onLongClick] or
+ * [onDoubleClick] won't be invoked
+ * @param matcher defines supported pointer types and required properties
+ * @param keyboardModifiers defines a condition that [PointerEvent.keyboardModifiers] has to match
+ * @param onLongClick will be called when user long presses on the element
+ * @param onDoubleClick will be called when user double clicks on the element
+ * @param onClick will be called when user clicks on the element
+ */
+@ExperimentalFoundationApi
+fun Modifier.onClick(
+    enabled: Boolean = true,
+    interactionSource: MutableInteractionSource,
```

Comment:
Similar to `combinedClickable`, we should handle the case when the composable is disposed (send a cancel in DisposableEffect)